### PR TITLE
[ROCm][Quantization] Add asymmetric INT8 quantization support to TritonInt8ScaledMMLinearKernel

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
- Adds asymmetric (AZP) INT8 quantization support to `TritonInt8ScaledMMLinearKernel`, which previously only supported symmetric input quantization.
- This unblocks asymmetric INT8 models (e.g. W8-Channel-A8-Dynamic-Asym-Per-Token) on ROCm, where the Triton kernel is the only viable INT8 fallback when AITER is not enabled.
- The implementation follows the same math as `CutlassInt8ScaledMMLinearKernel`: precomputes `azp_adj` (weight column sums) during weight loading, then applies the zero-point correction `scale_a * scale_b * azp * azp_adj` after the Triton GEMM. Supports both static per-tensor and dynamic per-token asymmetric schemes.

## Test plan
- [x] Run lm-eval harness with `Meta-Llama-3-8B-Instruct-INT8-compressed-tensors-asym.yaml` on ROCm

cc @kenroche

### CODE REVIEW

File: vllm/model_executor/kernels/linear/scaled_mm/triton.py [hunk 1]
```diff
@@ -62,32 +60,93 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
...
+
+                scale = (range_max - range_min) / (int8_traits.max - int8_traits.min)
+                replace_parameter(
+                    layer,
+                    i_s_name,
+                    torch.nn.Parameter(scale, requires_grad=False),
+                )
+
+                # AZP loaded as int8 but used as int32
+                azp = (int8_traits.min - range_min / scale).to(dtype=torch.int32)
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply vllm/model_executor/kernels/linear/scaled_mm/triton.py:88]: This is copied directly from CutlassInt8ScaledMMLinearKernel. The identical code runs in production on CUDA today without the epsilon guard. A quantized model with constant activations is broken at the model level.

**Comment by AndreasKaratzas** [comment]
TG is passing.

**Review by tjtanaa — APPROVED** [review]
LGTM

File: .buildkite/lm-eval-harness/configs/models-small-rocm.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Qwen2.5-1.5B-Instruct.yaml
 Meta-Llama-3.2-1B-Instruct-INT8-compressed-tensors.yaml
+Meta-Llama-3-8B-Instruct-INT8-compressed-tensors-asym.yaml
```
Comment (tjtanaa) [inline .buildkite/lm-eval-harness/configs/models-small-rocm.txt:3]: @AndreasKaratzas why is this file renamed to be fp8, it contains model that are unquantized and int8 quantized. It is better to keep it as `.buildkite/lm-eval-harness/configs/models-small-rocm.txt`

Comment (AndreasKaratzas) [reply .buildkite/lm-eval-harness/configs/models-small-rocm.txt:3]: @tjtanaa I thought putting the fp8 there to indicate that it must be run on machines with gfx942+ capability (gfx942 and gfx950 for the time being). But I renamed it as you suggested :)

File: .buildkite/test-amd.yaml [hunk 3]
```diff
@@ -2690,6 +2690,24 @@ steps:
   - pytest -s -v evals/gsm8k/test_gsm8k_correctness.py --config-list-file=configs/models-small.txt
 
 
+- label: LM Eval Small Models (FP8) # TBD
```
Comment (tjtanaa) [inline .buildkite/test-amd.yaml]: @AndreasKaratzas sorry, can you do the same.  let's name `- label: LM Eval Small Models (MI325)` since this is a test group running on mi325.

File: .buildkite/lm-eval-harness/configs/models-small-rocm.txt (hunk 2 above)
Comment (tjtanaa) [reply .buildkite/lm-eval-harness/configs/models-small-rocm.txt:3]: In this case, a proper way to gate this should be in https://github.com/vllm-project/vllm/blob/main/.buildkite/lm-eval-harness/test_lm_eval_correctness.py instead of through naming. Since the test will be enabled on different hardware.

Comment (AndreasKaratzas) [reply .buildkite/lm-eval-harness/configs/models-small-rocm.txt:3]: @tjtanaa Done :) Will enable the tests too for safety.
